pkg/platform/kube/functionres/lazy.go

```diff
@@ -570,6 +607,106 @@ func (lc *lazyClient) waitFunctionDeploymentReadiness(ctx context.Context,
 	return errors.New("Function deployment is not ready yet"), ""
 }
 
+// getFunctionDeployment returns function's deployment
+func (lc *lazyClient) getFunctionDeployment(ctx context.Context, function *nuclioio.NuclioFunction) (*appsv1.Deployment, error) {
+	return lc.kubeClientSet.AppsV1().
+		Deployments(function.Namespace).
+		Get(ctx, kube.DeploymentNameFromFunctionName(function.Name), metav1.GetOptions{})
+}
+
+func (lc *lazyClient) getFunctionPods(ctx context.Context,
```

Comment:
See if there other places that do the same and use this function there.

Also, as a convention - move below the functions that call it (e.g. below `checkFunctionInitContainersDone`).

Reply:
@TomerShor I believe that is the only place